src/components/ModalWithForm.js

```diff
@@ -27,6 +32,14 @@ export default class ModalWithForm extends Modal {
     });
   }
 
+  setLoading() {
+    this._submitButton.textContent = "Saving...";
+  }
+
+  resetButton() {
+    this._submitButton.textContent = "Save";
+  }
+
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // Here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`

Reply:
I'm not sure how this would be used.  The inputs are made into an object with the _getInputValues method, then this is passed to the event handler.  What exactly is "set" by setInputValues?  I use a method of the userInfo class to set the userInfo on the DOM.  I use _getUserInfo to create an object that is passed to the event handler.  I often hear of getter and setter methods, but my getInputValues method returns an inputObject--hence that is already "set."  The only setting after that is on the DOM, right?  ie userInfo class ... ?

Reply:
`setInputValues` can be used for setting the user data into the inputs on opening

You have this code in `index.js` for that

```js

  const currentUserInfo = userInfo.getUserInfo();
  profileNameInput.value = currentUserInfo.name;
  profileDescriptionInput.value = currentUserInfo.description;
```


If you implement `setInputValues`, then it will be


```js

  const currentUserInfo = userInfo.getUserInfo();
  profileEditPopup.setInputValues(currentUserInfo);  // here you insert the data into the inputs
```